integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala

```diff
@@ -689,6 +689,10 @@ object GlobalDictionaryUtil extends Logging {
           generatePredefinedColDictionary(colDictFilePath, table,
             dimensions, carbonLoadModel, sqlContext, hdfsLocation, dictfolderPath)
         }
+        if (headers.length > df.columns.length) {
+          logError("Either delimiter or fileheader provided is not correct")
```

Comment:
The log can be more informative like
" The number of columns in the file header do not match the number of columns in the data file; Either delimiter or fileheader provided is not correct"